spring-batch-infrastructure/src/main/java/org/springframework/batch/item/database/JdbcPagingItemReader.java

```diff
@@ -330,11 +330,27 @@ protected void doJumpToPage(int itemIndex) {
 		public T mapRow(ResultSet rs, int rowNum) throws SQLException {
 			startAfterValues = new LinkedHashMap<String, Object>();
 			for (Map.Entry<String, Order> sortKey : queryProvider.getSortKeys().entrySet()) {
-				startAfterValues.put(sortKey.getKey(), rs.getObject(sortKey.getKey()));
+				startAfterValues.put(sortKey.getKey(), findObject(rs, sortKey.getKey()));
 			}
 
 			return rowMapper.mapRow(rs, rowNum);
 		}
+
+		private Object findObject(ResultSet rs, String columnName) throws SQLException {
+			//If there is a '.' in the column name, it could be that we have a table name
+			// qualifier which won't be included as part of the column name in the result set.
+			//Or it could be a column alias which happens to contain a '.', in which case
+			// it might be included in the result set with its alias name.
+
+			int separatorIndex = columnName.indexOf('.');
+			if (separatorIndex > 0) {
+				try {
+					return rs.getObject(columnName.substring(separatorIndex));
+				} catch (SQLException ignored) { }
```

Comment:
I'm not a fan of testing this every time like this.  Generating stack traces is pretty performance intensive.  A better option would be to allow a `Map<String, String>` to be configurable to map any fields to the result set field.  This allows it to be more flexible as well I think.